crypto/evp/e_aes.c

```diff
@@ -401,7 +408,8 @@ static int aesni_xts_init_key(EVP_CIPHER_CTX *ctx, const unsigned char *key,
          * This addresses Rogaway's vulnerability.
          * See comment in aes_xts_init_key() below.
          */
-        if (memcmp(key, key + bytes, bytes) == 0) {
+        if ((!allow_insecure_decrypt || enc)
+                && memcmp(key, key + bytes, bytes) == 0) {
```

Comment:
Existing issue and just a single drive-by comment, but surely this should be a constant-time `memcmp`. :-)

(Ditto throughout.)

Reply:
Maybe??

I'm having trouble thinking what's going to leak.  The number of common bytes of the two keys?

I'll update just to be sure.